rest_framework/fields.py

```diff
@@ -939,7 +945,12 @@ def __init__(self, **kwargs):
 
     def to_internal_value(self, data):
 
-        if isinstance(data, six.text_type) and len(data) > self.MAX_STRING_LENGTH:
+        data = smart_text(data).strip()
```

Comment:
No because we could have a string as input without having the localize attribute to True.

I'll propose a change doing all string related sanitization only if the input is a string.